File: javascript/matrix/matrix.js
export class Matrix {
  constructor(matrixDescriptor) {
    this.numericRows = matrixDescriptor.split('\n')
      .map(row => row.split(' ').map(Number)); 
  };

  get rows() {
    return this.numericRows;
  }

  get columns() {
    const numCols = this.numericRows[0].length;
    let cols = [];
    for (var i = 0; i < numCols; i++) {
      cols[i] = this.numericRows.map(row => row[i]);
    }
    return cols;
  }
}

File: ruby/matrix/matrix.rb
class Matrix
  attr_reader :rows, :columns

  def initialize(matrix_text)
    @rows = get_rows(matrix_text)
    @columns = @rows.transpose
  end

  private

  def get_rows(matrix_text)
    matrix_text.lines.map { |l| l.split.map(&:to_i) }
  end
end